pkg/util/files.go

```diff
@@ -98,28 +93,49 @@ func Untar(source string, dest string) error {
 			break
 		}
 		if err != nil {
-			return err
+			return fmt.Errorf("Error during read of tar archive %v, err: %v", source, err)
 		}
 
-		if file.Typeflag == tar.TypeDir {
-			err = os.Mkdir(filepath.Join(dest, file.Name), 0755)
-			if err != nil {
-				return err
+		// If we have an extractionDir and this doesn't match, skip it.
+		if !strings.HasPrefix(file.Name, extractionDir) {
+			continue
+		}
+		// Now transform the filename to skip the extractionDir
+		file.Name = strings.TrimPrefix(file.Name, extractionDir)
+
+		// If file.Name is now empty this is the root directory we want to extract, and need not do anything.
+		if file.Name == "" && file.Typeflag == tar.TypeDir {
+			continue
+		}
+
+		fullPath := filepath.Join(dest, file.Name)
+
+		// At this point only directories and block-files are handled. Symlinks and the like are ignored.
```

Comment:
I believe we can treat symlinks as regular files here, which is how they were handled previously. In my original testing, this seemed to copy the symlink as-is, as opposed to processing to recreate it as a valid symlink.